### PR TITLE
WelcomeScreen, HomePage and Single Product Page

### PR DESCRIPTION
**Pull request by alexWaligorski**

### CODE REVIEW

File: src/components/WelcomePage/WelcomePage.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+import { Link } from "react-router-dom";
+
+export default function WelcomePage() {
+  return (
+    <div>
```
Comment (woeishi) [inline src/components/WelcomePage/WelcomePage.js:6]: App.js already has a enclosing `<div classname="App">...<div>`.
Is this additional div here (and in the other pages really necessary? or would a fragment `<>...</>` be enough

File: src/features/api/apiSlice.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { createApi, fetchBaseQuery } from "@reduxjs/toolkit/query/react";
+
+export const apiSlice = createApi({
+  reducerPath: "api",
+  baseQuery: fetchBaseQuery({ baseUrl: "https://dummyjson.com" }),
+  endpoints: (builder) => ({
+    getProducts: builder.query({ query: () => "/products?limit=10" }),
+    getCategories: builder.query({ query: () => "/products/categories" }),
+    getSingleProduct: builder.query({ query: (id) => `products/${id}` }),
```
Comment (woeishi) [inline src/features/api/apiSlice.js:9]: is often just named 'getProduct' (singular) implying that the identifier is needed as argument

File: src/features/ShopHomePage.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+          <ul>
+            {categories.map((category) => (
+              <li key={category}>{category}</li>
+            ))}
+          </ul>
+        </>
+      )}
+      {products && (
+        <>
+          <h2>Product Categories</h2>
```
Comment (woeishi) [inline src/features/ShopHomePage.js:23]: wrong title

File: src/features/ShopHomePage.js [hunk 4]
```diff
@@ -0,0 +1,35 @@
+import React from "react";
+import { useGetCategoriesQuery, useGetProductsQuery } from "./api/apiSlice";
+import { Link } from "react-router-dom";
+
+export default function ShopHomePage() {
+  const { data: categories } = useGetCategoriesQuery();
```
Comment (woeishi) [inline src/features/ShopHomePage.js:6]: loading state?

**Review by woeishi — CHANGES_REQUESTED** [review]
### RTK Naming Convention
seems like the suggested naming changed between redux and rtk. see [examples](examples/query/react):
- api slice -> 'src/app/services/api.js' or even 'src/app/services/products.js'
- components in 'src/features/products, src/features/categories,...'

File: src/components/WelcomePage/WelcomePage.js (hunk 1 above)
Comment (alexWaligorski) [reply src/components/WelcomePage/WelcomePage.js:6]: Good point! Changed div to fragment here and in other components.

File: src/features/api/apiSlice.js (hunk 2 above)
Comment (alexWaligorski) [reply src/features/api/apiSlice.js:9]: Aright!

File: src/features/ShopHomePage.js (hunk 3 above)
Comment (alexWaligorski) [reply src/features/ShopHomePage.js:23]: fixed

File: src/features/ShopHomePage.js (hunk 4 above)
Comment (alexWaligorski) [reply src/features/ShopHomePage.js:6]: Check!

**Comment by alexWaligorski** [comment]
Link to current RTK examples in comment is broken. Can you provide me with a functioning one? ;) Otherwise: All requests accepted and incorporated

**Pull request merged**